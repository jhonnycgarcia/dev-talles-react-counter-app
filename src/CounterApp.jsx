import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCount] = useState(value);

    const increment = () => {
        // setCount(counter + 1)
        setCount((prevCount) => prevCount + 1)
    }

    const decrement = () => {
        setCount((prevCount) => prevCount - 1)
    }

    const reset = () => {
        setCount(value)
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <button onClick={increment}>+1</button>
            <button onClick={decrement}>-1</button>
            <button onClick={reset}>Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
    value: 0,
}
