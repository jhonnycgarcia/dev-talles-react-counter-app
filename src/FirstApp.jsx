
import PropTypes from 'prop-types';

const getResult = () => {
  return 'Hola'
}

export const FirstApp = ({ name, title, subtitle }) => {

  return (
    <>
      <h1>{getResult() + ' ' + title}</h1>
      <div>{subtitle}</div>
      <div>{name}</div>
    </>
  )
}

FirstApp.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: 'Pedro',
  subtitle: 'Este es el subtítulo por defecto',
  title: 'Título por defecto',
}