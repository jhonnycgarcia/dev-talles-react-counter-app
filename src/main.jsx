import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'

import './style.css'

// import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'


createRoot(document.getElementById('root')).render(
    <StrictMode>
        {/* <FirstApp title='Hola Mundo' /> */}
        <CounterApp value={10} />
    </StrictMode>,
)