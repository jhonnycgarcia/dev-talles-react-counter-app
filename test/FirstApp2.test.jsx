import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en FirstApp2.jsx', () => {

    const title = 'Hola Mundo';
    const subtitle = 'Test Subtitle';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={ title } />);
        expect( container ).toMatchSnapshot();
    });

    test('debe de mostrar el texto "Hola Mundo"', () => {
        render(<FirstApp title={ title } />);
        expect( screen.getByText( title ) ).toBeTruthy();
    });

    test('debe de mostrar el titulo "Hola Mundo" en un h1', () => {
        render(<FirstApp title={ title } />);
        expect( screen.getByRole('heading', { level: 1 }) ).toBeTruthy();
        expect( screen.getByRole('heading', { level: 1 }).textContent ).toBe( title );
    });

    test('debe de mostrar el subtÃ­tulo enviado como prop', () => {
        render(<FirstApp title={ title } subtitle={ subtitle } />);
        expect( screen.getByText( subtitle ) ).toBeTruthy();
    });

});