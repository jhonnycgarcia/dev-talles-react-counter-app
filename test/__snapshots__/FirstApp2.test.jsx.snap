// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp2.jsx debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Mundo
  </h1>
  <p>
    Este es el subt√≠tulo por defecto
  </p>
  <p>
    Pedro
  </p>
</div>
`;
